### PR TITLE
VYAG-19 Input tags added in Timeline Search 

### PR DESCRIPTION
**Pull request by krazziekay**
-> The re-rendering of timeline when adding or updating new data was changed from setGroups, setItems to DataSet.add(), .update(), .remove() and .clear().
-> Search option in timeline changed from input type text to input tags

### CODE REVIEW

File: frontend-new/src/js/components/allocations/search/index.js [hunk 1]
```diff
@@ -29,47 +30,71 @@ class TimelineSearch extends Component {
...
+    let searchQuery = '';
+
+    return new Promise( (resolve, reject) => {
+      this.state.searchTags.map( (tag) => {
+        searchQuery += `${tag.type}=${tag.sid}&`;//concatenating the query parameters according to the tags in the search bar
+      });
+      AllocationService.fetchAllocationBySearch(searchQuery)
+      .then( (res) => {
+        if(res.data.data.length < 1) {
+          reject('No members for this project');
```
Comment (pratishshr) [inline frontend-new/src/js/components/allocations/search/index.js]: add a return here.
the function will not terminate after reject
so the resolve(res.data.data) will run as well.

File: frontend-new/src/js/components/allocations/search/index.js [hunk 2]
```diff
@@ -87,54 +112,95 @@ class TimelineSearch extends Component {
...
-      this.props.searchResults(res.data.data);
+  /**
+   * Input tag handlers
+   */
+  handleFilterChange =  (textInputValue, possibleSuggestionsArray) => {
+    let lowerCaseQuery = textInputValue.toLowerCase();
+    //Getting the possible suggestions list from the state 
+    possibleSuggestionsArray = this.state.searchSuggestions.map( (suggestion) => suggestion.text);
+    
+    return possibleSuggestionsArray.filter(function(suggestion)  {
```
Comment (pratishshr) [inline frontend-new/src/js/components/allocations/search/index.js]: use ES6 arrow functions

File: frontend-new/src/js/components/allocations/search/index.js [hunk 3]
```diff
@@ -87,54 +112,95 @@ class TimelineSearch extends Component {
...
+    tags.splice(i, 1);
+    this.setState({searchTags: tags});
+    //Check if the tags are empty or not
+    if(this.state.searchTags.length < 1) {
+      this.props.clearSearchField(true);
+      
+      return;
+    }
+
+    //Getting the allocation based on the tags of the search line and passing it to the parent,ie timeline
```
Comment (romitamgai) [inline frontend-new/src/js/components/allocations/search/index.js]: Remove inline comments

File: frontend-new/src/js/components/allocations/timeline/index.js [hunk 4]
```diff
@@ -72,8 +73,8 @@ class Timeline extends Component {
       horizontalScroll: true,
       verticalScroll: true,
       groupOrder: 'content',//for the order in the employee groups of the timeline
-      start: moment(moment()).startOf('month'),//the first day of the current week
-      end: moment(moment()).endOf('month'),//the last day of the current week
+      start: moment(moment()).subtract(1,'weeks').startOf('isoweek'),
+      end: moment(moment()).add(2, 'weeks').endOf('isoweek'),
```
Comment (pratishshr) [inline frontend-new/src/js/components/allocations/timeline/index.js]: moment(moment()).add(2, 'weeks').endOf('isoweek')
moment().add(2, 'weeks').endOf('isoweek')

Will this give the same result or different?

**Pull request merged**